src/auxiliary/t8code.jl

```diff
@@ -0,0 +1,452 @@
+    return nothing
+  end
+
+  # Initialize `t8code` with log level ERROR to prevent a lot of output in AMR simulations
+  t8_init(SC_LP_ERROR)
+
+  return nothing
+end
+
+@inline t8_mpi_comm() = mpi_comm().val
+
+function trixi_t8_unref_forest(forest)
+  t8_forest_unref(Ref(forest))
+end
+
+function t8_free(ptr)
+  sc_free(t8_get_package_id(), ptr)
+end
+
+function finalize_t8code()
+  # sc_finalize()
+  return nothing
+end
+
+function trixi_t8_count_interfaces(forest)
+  # Check that forest is a committed, that is valid and usable, forest.
+  @T8_ASSERT (t8_forest_is_committed(forest) != 0)
+
+  # Get the number of local elements of forest.
+  num_local_elements = t8_forest_get_local_num_elements(forest)
+  # Get the number of ghost elements of forest.
+  num_ghost_elements = t8_forest_get_num_ghosts(forest)
+  # Get the number of trees that have elements of this process.
+  num_local_trees = t8_forest_get_num_local_trees(forest)
+
+  current_index = t8_locidx_t(0)
+
+  local_num_conform = 0
+  local_num_mortars = 0
+  local_num_boundary = 0
+
+  for itree = 0:num_local_trees-1
+    tree_class = t8_forest_get_tree_class(forest, itree)
+    eclass_scheme = t8_forest_get_eclass_scheme(forest, tree_class)
+
+    # Get the number of elements of this tree.
+    num_elements_in_tree = t8_forest_get_tree_num_elements(forest, itree)
+
+    for ielement = 0:num_elements_in_tree-1
+      element = t8_forest_get_element_in_tree(forest, itree, ielement)
+
+      level = t8_element_level(eclass_scheme, element)
+
+      num_faces = t8_element_num_faces(eclass_scheme,element)
+
+      for iface = 0:num_faces-1
+
+        pelement_indices_ref = Ref{Ptr{t8_locidx_t}}()
+        pneighbor_leafs_ref = Ref{Ptr{Ptr{t8_element}}}()
+        pneigh_scheme_ref = Ref{Ptr{t8_eclass_scheme}}()
+
+        dual_faces_ref = Ref{Ptr{Cint}}()
+        num_neighbors_ref = Ref{Cint}()
+
+        forest_is_balanced = Cint(1)
+
+        t8_forest_leaf_face_neighbors(forest,itree,element,
+          pneighbor_leafs_ref, iface, dual_faces_ref, num_neighbors_ref,
+          pelement_indices_ref, pneigh_scheme_ref, forest_is_balanced)
+
+        num_neighbors      = num_neighbors_ref[]
+        neighbor_ielements = unsafe_wrap(Array,pelement_indices_ref[],num_neighbors)
+        neighbor_leafs     = unsafe_wrap(Array,pneighbor_leafs_ref[],num_neighbors)
+        neighbor_scheme    = pneigh_scheme_ref[]
+
+        if num_neighbors > 0
+          neighbor_level = t8_element_level(neighbor_scheme, neighbor_leafs[1])
+
+          # Conforming interface: The second condition ensures we only visit the interface once.
+          if level == neighbor_level && current_index <= neighbor_ielements[1]
+          # TODO: Find a fix for the case: Single element on root level with periodic boundaries.
+          # elseif level == neighbor_level && 
+          #   (all(Int32(current_index) .< neighbor_ielements) || 
+          #   level == 0 && (iface == 0 || iface == 2 || iface == 4))
```

Comment:
Still open?

Reply:
Yes. Still an open issue.

Reply:
Do you have a list of open issues with this in Trixi.jl? If not, please create one, add the TODO notes there, and resolve this comment (and similar ones)